crates/ruff_python_semantic/src/binding.rs

```diff
@@ -266,6 +275,14 @@ impl<'a> Binding<'a> {
             .map(|statement_id| semantic.statement(statement_id))
     }
 
+    /// Returns the parent of the expression in which the binding was defined.
+    ///
+    /// This is only really applicable for assignment expressions.
+    pub fn parent_expression<'b>(&self, semantic: &SemanticModel<'b>) -> Option<&'b ast::Expr> {
+        self.source
+            .and_then(|expression_id| semantic.parent_expression(expression_id))
+    }
+
```

Comment:
Should this method be just `expression` instead? Considering `assert (x := 1)`, for the binding `x` this would return the named expression, right? If so, shouldn't that be the "expression in which the binding was defined" (similar to the `statement` method)?

Reply:
Yeah, that makes sense. I named it like this because the implementation is more similar to `Binding::parent_statement()` than to `Binding::statement()`. That's because `self.source.and_then(|expression_id| semantic.expression(expression_id))` will give you the `ast::Expr::Name()` that represents the symbol `x` rather than the `ast::Expr::Named()` that represents the `x := 1` walrus. But I don't think you ever really want the `ast::Expr::Name` node (you can get the range directly from the `Binding`), so your naming is likely more intuitive for users of the API.